### PR TITLE
PMTiles Show: Generate an encoded URL that will quickly take you to PMTiles.io

### PR DESCRIPTION
**Pull request by roblabs**
#### Minor feature enhancement to `pmtiles show`

* `pmtiles show` now generates an encoded URL that will quickly take you to [PMTiles.io](http://pmtiles.io/) with your archive passed in.
* Perfect for 1-click links for inspecting PMTiles.
* Easily clickable from Visual Studio code or on macOS use `open -a <browser_name>`
* Future ideas below.

---

#### Test on S3 compatible PMTile

```bash
# log an element "xray:"
pmtiles show https://demo-bucket.protomaps.com/v4.pmtiles
```

Result

* Output looks something like:

    pmtiles spec version: 3
    tile type: mvt
    ...
    xray: https://pmtiles.io/#url=https%3A%2F%2Fdemo-bucket.protomaps.com%2Fv4.pmtiles

---

#### Test on local PMTile

For the case of a local file, the "xray: " log element is disabled.

```bash
# Show works as before
pmtiles show pmtiles/fixtures/test_fixture_1.pmtiles
```

---
##### `tile`
- [ ] @bdon's idea from Slack:  "*allow `pmtiles tile` to link to the tile inspector*"
  * https://pmtiles.io/tile/#zxy=1/0/0&url=https%3A%2F%2Fdemo-bucket.protomaps.com%2Fv4.pmtiles

##### local file + `serve`
- [ ] `pmtiles show` for a local file:   "*Maybe provide a hint that the Go tool can also serve a folder*":  
  * `pmtiles serve pmtiles/fixtures`
- [ ]  `pmtiles serve` could also log a link to the locally servedTileJSON:  `https://pmtiles.io/#url=http%3A%2F%2Flocalhost%3A8080%2Ftest_fixture_1.json`
  * Works with:  `pmtiles serve pmtiles/fixtures --public-url=http://localhost:8080`

### CODE REVIEW

File: pmtiles/show.go [hunk 1]
```diff
@@ -109,6 +111,10 @@ func Show(_ *log.Logger, output io.Writer, bucketURL string, key string, showHea
 					fmt.Println(k, "<object...>")
 				}
 			}
+
+			if strings.HasPrefix(bucketURL, "http") {
```
Comment (bdon) [inline pmtiles/show.go:115]: this may actually need to be `https://` since non-SSL resources cannot be securely previewed on https://pmtiles.io (which uses SSL).

File: pmtiles/show.go [hunk 2]
```diff
@@ -109,6 +111,10 @@ func Show(_ *log.Logger, output io.Writer, bucketURL string, key string, showHea
...
+				fmt.Println("xray: https://pmtiles.io/#url=" + url.QueryEscape(bucketURL+"/"+key))
```
Comment (bdon) [inline pmtiles/show.go:116]: What if we use `web viewer:` instead of `xray:` to make it clearer that viewing the file requires an internet connection (the pmtiles CLI is otherwise self-contained and works offline)

File: pmtiles/show.go (hunk 1 above)
Comment (roblabs) [reply pmtiles/show.go:115]: The `http` prefix on line 115 works with Vercel `serve`.   Test case (maybe a corner case, so let me know your thoughts):

```bash
# from https://www.npmjs.com/package/serve
npm i serve.  

# looking for: protomaps(vector)ODbL_firenze.pmtiles
cd PMTiles/spec/v3
serve . --cors -p 3857

# with quotes because of ()
pmtiles show "http://localhost:3857/protomaps(vector)ODbL_firenze.pmtiles"
```

Will generate a working link to PMTiles.io:  
https://pmtiles.io/#url=http%3A%2F%2Flocalhost%3A3857%2Fprotomaps%28vector%29ODbL_firenze.pmtiles

*Firefox.   Use case for `pmtiles show`:  easily view many `*.pmtiles` stored in a folder served by Vercel `serve`*

File: pmtiles/show.go (hunk 2 above)
Comment (roblabs) [reply pmtiles/show.go]: Resolved with 10479888bc41fe18a685938644b03b6c9c3dec1e

File: pmtiles/show.go (hunk 1 above)
Comment (bdon) [reply pmtiles/show.go:115]: Can you try in a non-Firefox browser?

File: pmtiles/show.go [hunk 3]
```diff
@@ -109,6 +111,10 @@ func Show(_ *log.Logger, output io.Writer, bucketURL string, key string, showHea
 					fmt.Println(k, "<object...>")
 				}
 			}
+
+			if strings.HasPrefix(bucketURL, "https://") {
```
Comment (roblabs) [inline pmtiles/show.go:115]: Works in Safari & Firefox for macOS

File: pmtiles/show.go (hunk 1 above)
Comment (roblabs) [reply pmtiles/show.go]: Using Vercel `serve` from a local server instance fails on Safari.  I have updated [L15](https://github.com/protomaps/go-pmtiles/pull/229/files#diff-7724917e94154660f21df5b03e9929332bee1a575d01b3cd141c8de5542536b0R115) to only generated URL encoding with `https://`

**Review by bdon — APPROVED** [review]

**Comment by bdon** [comment]
Released in https://github.com/protomaps/go-pmtiles/releases/tag/v1.28.0
